var/spack/repos/builtin/packages/py-tensorflow/package.py

```diff
@@ -300,6 +309,7 @@ class PyTensorflow(Package, CudaPackage):
     patch('contrib_cloud_1.9.patch', when='@1.9')
     patch('contrib_cloud_1.4.patch', when='@1.4:1.8')
     patch('contrib_cloud_1.1.patch', when='@1.1:1.3')
+    patch('tensorflow_rocm.patch', when='+rocm')
```

Comment:
Does this patch work for all versions of TF? If not, you'll need to use multiple patches.

Reply:
i have just added the patch for the current version. did not test the previous versions.

Reply:
This should have a `when=` clause if it only works with a particular version.